### PR TITLE
OBD2 CAN broken due to math base mismatch 

### LINKED ISSUE #6851: OBD2 CAN broken due to math base mismatch
**Describe the bug**
OBD2 thinger asks:

`     5)     76967.7  Rx         07DF  8  02 01 20 00 00 00 00 00`
show me supported pids 0x21 - 0x40

`     6)     76968.0  Rx         07E8  8  06 41 20 00 01 00 00 00`

we respond with 
`00 01 00 00 `

when we should respond with

`10 00 00 00`

this is because this piece of code ([firmware/controllers/can/obd2.cpp](https://github.com/rusefi/rusefi/blob/master/firmware/controllers/can/obd2.cpp#L110)) sends  bitOffset as decimal, instead of hex., first one works
2nd and 3rd don't

```

        case PID_SUPPORTED_PIDS_REQUEST_01_20:
                obdWriteSupportedPids(pid, 1, supportedPids0120, busIndex);
                break;
        case PID_SUPPORTED_PIDS_REQUEST_21_40:
                obdWriteSupportedPids(pid, 21, supportedPids2140, busIndex);
                break;
        case PID_SUPPORTED_PIDS_REQUEST_41_60:
                obdWriteSupportedPids(pid, 41, supportedPids4160, busIndex);
                break;

```
```
static void obdWriteSupportedPids(int PID, int bitOffset, const int16_t *supportedPids, size_t busIndex);
.
.
.
                value |= 1 << (31 + bitOffset - supportedPids[i]);
.
.
```

it should be

`                obdWriteSupportedPids(pid, 0x21, supportedPids2140, busIndex);`

and 

`                obdWriteSupportedPids(pid, 0x41, supportedPids4160, busIndex);`

**Hardware**
 - Board: doesn't matter

**To Reproduce**
1. Hook up OBD2 hud
2. see hud not get AFR
3. wonder why ?

**Expected behavior**
We are supposed to tell obd2 thing that we support 0x24 pid, which is lambda1 but we don't tell it that because of a math issue

### PR DESCRIPTION
**Pull request by FDSoftware**
resolves #6851
- updated can_msg_tx/obd2 code for unit tests
-  fixed supported pid error
- WIP unit tests on obd2

### CODE REVIEW

**Comment by rusefillc** [comment]
Love the unit tests!!!

**Comment by FDSoftware** [comment]
I'm having a problem with the tests, which compile and run well locally, but not on the server. :cry: 

 make && ./build/rusefi_test --gtest_filter="CanObd*"
 
```
Compiling test_math.cpp
Compiling gtest-all.cpp
Compiling gmock-all.cpp
rm -rf build/obj/*gcda
Linking build/rusefi_test
Note: Google Test filter = CanObd*
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from CanObd2
[ RUN      ] CanObd2.obdSendPacket
commonInitEngineController
doInitElectronicThrottle No
getSpeedDensityAirmass map=96.26
OBD Sending CAN bus0 message: ID=7e8/l=8 3 40 c ff 0 0 0 0 
Not enough data for unittest_obdSendPacket.logicdata
[       OK ] CanObd2.obdSendPacket (0 ms)
[----------] 1 test from CanObd2 (0 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (0 ms total)
[  PASSED  ] 1 test.
DONE returning 0
All triggers exported to triggers.txt
```

**Comment by rusefillc** [comment]
@FDSoftware allow me to clarify/hint a bit: "since test runs fine locally while all tests together fail on the server"

hinting at single vs all

**Comment by FDSoftware** [comment]
hmm it only happens if I run them without the filter, I think it's holding some txCanBuffer data from another CAN related test

File: firmware/hw_layer/drivers/can/can_msg_tx.cpp [hunk 1]
```diff
@@ -59,9 +59,25 @@ CanTxMessage::CanTxMessage(CanCategory p_category, uint32_t eid, uint8_t dlc, si
 #endif // HAS_CAN_FRAME
 }
 
-CanTxMessage::~CanTxMessage() {
-#if EFI_SIMULATOR
+CanTxMessage::~CanTxMessage(){
```
Comment (rusefillc) [inline firmware/hw_layer/drivers/can/can_msg_tx.cpp:62]: i like that space

considering how amazing this PR is I am fine loosing the space

Comment (FDSoftware) [reply firmware/hw_layer/drivers/can/can_msg_tx.cpp]: fixed, it was vscode's auto-formatting, do we have any specific tool to format the files?

**Comment by rusefillc** [comment]
@FDSoftware https://github.com/rusefi/rusefi/blob/master/.editorconfig is supposed to be doing something

**Comment by rusefillc** [comment]
Nice, it's green! What's the next step? Merge or waiting for @ggurov to test branch artifact?

**Comment by FDSoftware** [comment]
Yes!, If ggrouv can test it with the same setup that would be ideal; I also want to add a few more tests, I have a couple of functions and 4-5 cases of handleGetDataRequest without coverage. Today at the end of the day or tomorrow I will be able to finish the tests that I am missing.

**Comment by rusefillc** [comment]
@ggurov needs it in master to test with less effort

**Pull request merged**